addons/waterways/gui/river_gizmo.gd

```diff
@@ -253,24 +253,24 @@ func _set_handle(gizmo: EditorNode3DGizmo, index: int, secondary: bool, camera:
 			var ray_to = ray_from + (ray_dir * AXIS_CONSTRAINT_LENGTH)
 			var result = Geometry3D.get_closest_points_between_segments(axis_from, axis_to, ray_from, ray_to)
 			new_pos = result[0]
-		
+
 		elif editor_plugin.constraint in PLANE_MAPPING:
 			var normal: Vector3 = PLANE_MAPPING[editor_plugin.constraint]
 			if editor_plugin.local_editing:
 				normal = _handle_base_transform.basis * (normal)
-			var projected : Vector3 = old_pos_global.project(normal)
-			var direction : Vector3 = sign(projected.dot(normal))
-			var distance : Vector3 = direction * projected.length()
+			var projected := old_pos_global.project(normal)
+			var direction := signf(projected.dot(normal))
+			var distance := direction * projected.length()
```

Comment:
You definitely should not remove the explicit static typing. At a glance do you know the return types of `project()`, `dot()` and `length()`? Does everyone else?

Reply:
Well, the problem was caused by using `sign()` function which returns variant and assignment to `direction: Vector3` failed because `sign()` *chose* to use the float-variant. Using `signf()` enforced the float version, `:=` assignment fixed the type mismatch.

> At a glance do you know the return types of project(), dot() and length()? Does everyone else?

When I frequently use those, yes. When fn unnaturally ends with `i` or `f`, you know it's int/float. When I rarely use them, I can ctrl+click them to see the definition in help, which is suboptimal (on-hover prototype would be better but it's not there yet), but works. Using `:=` helps avoiding variant stuff because doing so will result in parse error. But that's my rationale from my code style and I'll happily change it if project maintainer says so.

Reply:
> Using signf() enforced the float version, := assignment fixed the type mismatch.

Changing it to get it working is fine, so you should type it as float. You shouldn't reduce the maintainer's typing as done here by stripping out all the vec3s.